examples/aws-kms-key-complete/main.tf

```diff
@@ -0,0 +1,69 @@
+resource "aws_kms_key" "key" {
+  description = "MongoDB Atlas KMS key."
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Id" : "key-default-1",
+    "Statement" : [
+      {
+        "Sid" : "Allow administration of the key",
+        "Effect" : "Allow",
+        "Principal" : {
+          "AWS" : "arn:aws:iam::{AWS_ACCOUNT}:root"
+        },
+        "Action" : [
+          "kms:*"
+        ],
+        "Resource" : "*"
+      },
+      {
+        "Sid" : "Allow use of the key by specific IAM user",
+        "Effect" : "Allow",
+        "Principal" : {
+          "AWS" : module.aws-kms-key.role_arn
```

Comment:
question: since there is a cyclic dependency, would this example be able to run? If not possible, would keep the creation of the `aws_kms_key` more simplified and have the readme provide details that policy of kms can be adjusted with fine grained permissions to the created role of the module.

Reply:
The example can run without any problem. What I suspect is that terraform starts creating the kms_key resource, but as one of the attributes is using the arn of the iam role, does not finish until the iam role resource has been created in the module. In this case there is no cyclic dependency because the ARN of the aws_kms_key is used in the aws_iam_role_policy resource, not in the aws_iam_role (and later in the mongodbatlas_encryption_at_rest_resource).